store/user/dashboard.js

```diff
@@ -4,12 +4,21 @@ class DashboardUser extends User {
   static defaultValues = {
     ...User.defaultValues,
     dataProviders: [],
+    dataProvider: null,
```

Comment:
~You don't change the data provider here, do you?~ You do, I see.

It's kind of okay because you check access rights in the setter, what is good. However, storing the parent's data in the child object does not feel good. You can add a private variable to the `Root` or rethink data provider changing action.

Reply:
> However, storing the parent's data in the child object does not feel good.

That's why I wanted to have everything in user entity.

Reply:
You can if you want to. JavaScript allows it. However, I would move it to the parent later.